src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java

```diff
@@ -302,18 +309,22 @@ public void onFinalized(Run run) {
                 buildData = new BuildData(run, null);
             } catch (IOException | InterruptedException e) {
                 DatadogUtilities.severe(logger, e, "Failed to parse finalized build data");
+                cleanUpTraceActions(run);
```

Comment:
Isn't the `finally` below called regardless of the return in this block? That would mean cleaning twice unnecessarily

Reply:
Totally correct. I'll remove this.